src/main/java/org/testng/internal/TestResult.java

```diff
@@ -33,33 +38,49 @@
   private int parameterIndex;
   private boolean m_wasRetried;
 
-  public TestResult() {}
+  private TestResult() {
+    //defeat instantiation. We have factory methods.
+  }
+
+  public static TestResult newEmptyTestResult() {
+    return new TestResult();
+  }
+
+  public static TestResult newTestResultFor(ITestNGMethod method) {
+    return newContextAwareTestResult(method, null);
+  }
 
-  public TestResult(ITestNGMethod method, Throwable throwable, ITestContext context) {
+  public static TestResult newContextAwareTestResult(ITestNGMethod method, ITestContext ctx) {
+    TestResult result = newEmptyTestResult();
     long time = System.currentTimeMillis();
-    init(method.getTestClass(), method, throwable, time, time, context);
+    result.init(method, null, time, 0L, ctx);
+    return result;
   }
 
-  public TestResult(
-      IClass testClass,
-      ITestNGMethod method,
-      Throwable throwable,
-      long start,
-      long end,
-      ITestContext context) {
-    init(testClass, method, throwable, start, end, context);
+  public static TestResult newTestResultWithCauseAs(ITestNGMethod method, ITestContext ctx,
+      Throwable t) {
+    TestResult result = newEmptyTestResult();
+    long time = System.currentTimeMillis();
+    result.init(method, t, time, time, ctx);
+    return result;
+  }
+
+  public static TestResult newEndTimeAwareTestResult(ITestNGMethod method, ITestContext context,
+      Throwable throwable, long start) {
+    TestResult result = newEmptyTestResult();
+    long time = System.currentTimeMillis();
+    result.init(method, throwable, start, time, context);
+    return result;
   }
 
   public void init(

```

Comment:
@juherr - That is currently not possible. This method is being used outside.  There is currently one use case from within `org.testng.internal.Invoker#invokeMethod` wherein instantiation and initialisation is being done in two separate explicit calls. So refactoring this would need additional changes and that bloats my changeset. I am happy to do this as a separate PR.

Reply:
Could you add a TODO or open an issue for that?

Reply:
I have raised an issue for this https://github.com/cbeust/testng/issues/1893